pkg/changelog/changelog_writer.go

```diff
@@ -16,25 +16,27 @@ type (
 		WriteLine WriteLine
 	}
 	ChangelogEntry struct {
-		Seq    int64
-		Family string
-		Table  string
-		Key    []interface{}
+		Seq         int64
+		Family      string
+		Table       string
+		Key         []interface{}
+		LedgerSeq   schema.DMLSequence
+		Transaction bool
 	}
 )
 
-func NewChangelogEntry(seq int64, family string, table string, key []interface{}) *ChangelogEntry {
-	return &ChangelogEntry{Seq: seq, Family: family, Table: table, Key: key}
-}
-
 func (w *ChangelogWriter) WriteChange(e ChangelogEntry) error {
 	structure := struct {
```

Comment:
I don't really get why we inline this struct definition instead of just putitng the json tags on the ChangeLogEntry.

Reply:
This is a good question -- I don't get it either.

It reminds me that I'm unsure about why we have 2 separate structs for the changelog entries. Felt like I was having to sprinkle the same fields all over the place to get it through the pipes.
* writing (this code): https://github.com/segmentio/ctlstore/blob/365206507ae8ef2f4a70bf416487a22bb2b0803a/pkg/changelog/changelog_writer.go#L30-L36
* reading: https://github.com/segmentio/ctlstore/blob/365206507ae8ef2f4a70bf416487a22bb2b0803a/pkg/event/entry.go#L3-L11

Reply:
I wonder if that's a consequence of calling the root of this `pkg` vs `internal`

Reply:
Feels like there should be just one definition of this data.

Reply:
With the json annotations too.